### PR TITLE
Add restful api support.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**

### CODE REVIEW

File: cmd/pd-server/main.go [hunk 1]
```diff
@@ -33,6 +34,7 @@ var (
 	rootPath        = flag.String("root", "/pd", "pd root path in etcd")
 	leaderLease     = flag.Int64("lease", 3, "leader lease time (second)")
 	logLevel        = flag.String("L", "debug", "log level: info, debug, warn, error, fatal")
+	httpAddr        = flag.String("httpAddr", ":9090", "http server listening address")
```
Comment (siddontang) [inline cmd/pd-server/main.go]: http-addr

File: server/server.go [hunk 2]
```diff
@@ -28,6 +28,9 @@ import (
 	statsd "gopkg.in/alexcesaro/statsd.v2"
 )
 
+// PdServer is global pd server.
+var PdServer *Server
```
Comment (siddontang) [inline server/server.go]: why using global var? I don't think it is a good habit.

File: server/api/server.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+func ServeHTTP(addr string, srv *server.Server) {
+	engine := negroni.New()
+
+	recovery := negroni.NewRecovery()
+	engine.Use(recovery)
+
+	router := createRouter(srv)
+	engine.UseHandler(router)
+
+	http.ListenAndServe(addr, engine)
```
Comment (disksing) [inline server/api/server.go]: If `ListenAndServe` fails, need return the error or log it down.

File: server/balancer_worker.go [hunk 4]
```diff
@@ -151,6 +159,26 @@ func (bw *balancerWorker) getBalanceOperator(regionID uint64) *balanceOperator {
 	return bw.balanceOperators[regionID]
 }
 
+func (bw *balancerWorker) getBalanceOperators() map[uint64]*balanceOperator {
+	bw.RLock()
+	defer bw.RUnlock()
+
+	return bw.balanceOperators
```
Comment (disksing) [inline server/balancer_worker.go]: Need clone?

File: server/balancer_worker.go [hunk 5]
```diff
@@ -203,3 +231,41 @@ func (bw *balancerWorker) doBalance() error {
 	log.Info("find no proper region for balance, retry later")
 	return nil
 }
+
+func (bw *balancerWorker) mockBalanceOperator() {
```
Comment (disksing) [inline server/balancer_worker.go]: What is it used for? Maybe move it into a _test file?

Comment (qiuyesuifeng) [inline server/balancer_worker.go]: It is convenient for us to have a test.
How about moving it to a new file called `mock.go`?

File: server/api/server.go (hunk 3 above)
Comment (qiuyesuifeng) [inline server/api/server.go]: Yes, maybe we should exit if we cannot start a http service.

**Comment by qiuyesuifeng** [comment]
@disksing  I will address comments after #164 is merged.

**Comment by qiuyesuifeng** [comment]
Close it, reopen a new PR.

**Pull request closed without merging**